From 968cf201800dde620907f7217cb84b2f2cba0711 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Gapi=C5=84ski?= <mike@gapinski.eu>
Date: Wed, 1 May 2024 17:07:32 +0000
Subject: [PATCH] write bootloader-update image to vendor_dlkm

Rebase the Tesla Android updater payload onto the Android 15 GloDroid
kernel makefile layout.
---
 platform/kernel/kernel.mk | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/platform/kernel/kernel.mk b/platform/kernel/kernel.mk
index 3f027ad..7d77a96 100644
--- a/platform/kernel/kernel.mk
+++ b/platform/kernel/kernel.mk
@@ -66,12 +66,25 @@ ANDROID_DTS_OVERLAY	?= $(LOCAL_PATH)/empty.dts
 
 DTB_OUTPUT		:= $(PRODUCT_OUT)/dtb.img
 
+RPI_FIRMWARE_DIR := glodroid/bootloader/raspberry-fw
+DTBS_DIR      := $(KERNEL_OUT)/install/dtbs
+
 $(TARGET_VENDOR_MODULES)/modules.dep: $(KERNEL_TARGET)
 	rm -rf $(TARGET_VENDOR_MODULES)/kernel
 	ls $(TARGET_VENDOR_MODULES)/modules.* | grep -v modules.blocklist | xargs rm -f
 	mkdir -p $(TARGET_VENDOR_MODULES)/kernel
 	cp -r $(KERNEL_OUT)/install/modules/lib/modules/GloDroid/kernel/* $(TARGET_VENDOR_MODULES)/kernel/
 	cp -r $(KERNEL_OUT)/install/modules/lib/modules/GloDroid/modules.* $(TARGET_VENDOR_MODULES)/
+	mkdir -p $(TARGET_OUT_VENDOR_DLKM)/boot/overlays
+	cp -r $(RPI_FIRMWARE_DIR)/boot/overlays $(TARGET_OUT_VENDOR_DLKM)/boot
+	cp -r $(RPI_FIRMWARE_DIR)/boot/bootcode.bin $(TARGET_OUT_VENDOR_DLKM)/boot/bootcode.bin
+	cp -r $(RPI_FIRMWARE_DIR)/boot/start_x.elf $(TARGET_OUT_VENDOR_DLKM)/boot/start_x.elf
+	cp -r $(RPI_FIRMWARE_DIR)/boot/start4x.elf $(TARGET_OUT_VENDOR_DLKM)/boot/start4x.elf
+	cp -r $(RPI_FIRMWARE_DIR)/boot/fixup_x.dat $(TARGET_OUT_VENDOR_DLKM)/boot/fixup_x.dat
+	cp -r $(RPI_FIRMWARE_DIR)/boot/fixup4x.dat $(TARGET_OUT_VENDOR_DLKM)/boot/fixup4x.dat
+	cp -r $(DTBS_DIR)/broadcom/bcm2711-rpi-4-b.dtb $(TARGET_OUT_VENDOR_DLKM)/boot/bcm2711-rpi-4-b.dtb
+	cp -r $(DTBS_DIR)/broadcom/bcm2711-rpi-400.dtb $(TARGET_OUT_VENDOR_DLKM)/boot/bcm2711-rpi-400.dtb
+	cp -r $(DTBS_DIR)/broadcom/bcm2711-rpi-cm4.dtb $(TARGET_OUT_VENDOR_DLKM)/boot/bcm2711-rpi-cm4.dtb
 	touch $@
 
 $(PRODUCT_OUT)/kernel: $(KERNEL_TARGET) $(TARGET_VENDOR_MODULES)/modules.dep
-- 
2.51.0
